astropy/nddata/blocks.py

```diff
@@ -0,0 +1,206 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+This module includes helper functions for array operations.
+"""
+
+import numpy as np
+
+from .decorators import support_nddata
+
+__all__ = ['reshape_as_blocks', 'block_reduce', 'block_replicate']
+
+
+def _process_block_inputs(data, block_size):
+    data = np.asanyarray(data)
+
+    block_size = np.atleast_1d(block_size)
+    if data.ndim > 1 and len(block_size) == 1:
+        block_size = np.repeat(block_size, data.ndim)
+
+    if len(block_size) != data.ndim:
+        raise ValueError('block_size must be a scalar or have the same '
+                         'length as the number of data dimensions')
+
+    if np.any(block_size <= 0):
+        raise ValueError('block_size elements must be strictly positive')
+
+    block_size_int = block_size.astype(int)
+    if np.any(block_size_int != block_size):  # e.g., 2.0 is OK, 2.1 is not
+        raise ValueError('block_size elements must be integers')
+
+    return data, block_size_int
+
+
+def reshape_as_blocks(data, block_size):
+    """
+    Reshape a data array into blocks.
```

Comment:
This needs a "version added" directive?

Reply:
I can add that, but I didn't think astropy used it very much.

Reply:
Its usage is not consistent throughout the package, but doesn't hurt?

Reply:
Done.

Reply:
>  I didn't think astropy used it very much

not yet, but hopefully we change that 😅